src/server.ts

```diff
@@ -135,49 +141,85 @@ export class Server {
    * @example
    * const contractId = "CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5";
    * const key = xdr.ScVal.scvSymbol("counter");
-   * server.getContractData(contractId, key).then(data => {
+   * server.getContractData(contractId, key, 'temporary').then(data => {
    *   console.log("value:", data.xdr);
    *   console.log("lastModified:", data.lastModifiedLedgerSeq);
    *   console.log("latestLedger:", data.latestLedger);
    * });
    *
-   * Allows you to directly inspect the current state of a contract. This is a backup way to access your contract data which may not be available via events or simulateTransaction.
+   * Allows you to directly inspect the current state of a contract. This is a
+   * backup way to access your contract data which may not be available via
+   * events or simulateTransaction.
    *
-   * @param {string} contractId - The contract ID containing the data to load. Encoded as Stellar Contract Address e.g. `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex string for backwards compatibility, but will likely be deprecated in the future.
+   * @param {string} contractId - The contract ID containing the data to load.
+   *    Encoded as Stellar Contract Address e.g.
+   *    `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex
+   *    string for backwards compatibility, but will likely be deprecated in the
+   *    future.
    * @param {xdr.ScVal} key - The key of the contract data to load.
-   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   * @param {string} [expirationType] - The "durability keyspace" that this
+   *    ledger key belongs to, which is either 'temporary' (the default) or
+   *    'persistent'.
+   *
+   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the
+   *    {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   *
+   * @warning If the data entry in question is a 'temporary' entry, it's
+   *    entirely possible that it has expired out of existence. Future versions
+   *    of this client may provide abstractions to handle that.
    */
   public async getContractData(
     contractId: string,
```

Comment:
Minor suggestion: Might be convenient if this also took a few more variants.
```suggestion
    contractId: string | Address | ScAddress,
```

Reply:
good idea, added `Contract | Address`. I don't think `xdr.ScAddress` is useful? and also hard to detect because you can't `instanceof` it reliably, I think